src/backend/render/mod.rs

```diff
@@ -799,23 +807,25 @@ where
         None => (0, 0).into(),
     };
 
-    elements.extend_from_workspace_elements(
-        workspace
-            .render::<R>(
-                renderer,
-                (!move_active && is_active_space).then_some(&last_active_seat),
-                overview,
-                resize_indicator,
-                active_hint,
-                theme,
-            )
-            .map_err(|_| OutputNoMode)?,
-        offset.to_physical_precise_round(output_scale),
-    );
+    if element_filter != ElementFilter::LayerShellOnly {
+        elements.extend_from_workspace_elements(
+            workspace
+                .render::<R>(
+                    renderer,
+                    (!move_active && is_active_space).then_some(&last_active_seat),
+                    overview,
+                    resize_indicator,
+                    active_hint,
+                    theme,
+                )
+                .map_err(|_| OutputNoMode)?,
+            offset.to_physical_precise_round(output_scale),
+        );
+    }
```

Comment:
Shouldn't the workspace elements from the "previous" workspace also be ignored the same? Or does this rely on animations being disabled for the overview?

Reply:
Yeah, it doesn't make a difference with the animations disabled, though it wouldn't hurt to add another if condition for that.